packages/mdc-menu/index.js

```diff
@@ -27,79 +29,105 @@ class MDCMenu extends MDCComponent {
   /** @param {...?} args */
   constructor(...args) {
     super(...args);
-    /** @private {!Element} */
-    this.previousFocus_;
+    /** @private {!MDCMenuSurface} */
+    this.menuSurface_;
+    /** @private {!MDCList} */
+    this.list_;
+    /** @private {!Function} */
+    this.handleKeydown_ = this.foundation_.handleKeydown.bind(this.foundation_);
+    /** @private {!Function} */
+    this.handleClick_ = this.foundation_.handleClick.bind(this.foundation_);
   }
 
   /**
-   * @param {!Element} root
+   * @param {!HTMLElement} root
    * @return {!MDCMenu}
    */
   static attachTo(root) {
     return new MDCMenu(root);
   }
 
+  initialize(
+    menuSurfaceFactory = (el) => new MDCMenuSurface(el),
+    listFactory = (el) => new MDCList(el)) {
+    this.menuSurface_ = menuSurfaceFactory(this.root_);
+
+    const list = this.root_.querySelector(strings.LIST_SELECTOR);
+    if (list) {
+      this.list_ = listFactory(list);
+      this.list_.wrapFocus = true;
+    }
+  }
+
+  destroy() {
+    if (this.list_) {
+      this.list_.destroy();
+    }
+
+    this.menuSurface_.destroy();
+    this.deregisterListeners_();
+    super.destroy();
+  }
+
   /** @return {boolean} */
   get open() {
-    return this.foundation_.isOpen();
+    return this.menuSurface_.open;
   }
 
   /** @param {boolean} value */
   set open(value) {
     if (value) {
-      this.foundation_.open();
+      this.show();
     } else {
-      this.foundation_.close();
+      this.hide();
     }
   }
 
-  /** @param {{focusIndex: ?number}=} options */
-  show({focusIndex = null} = {}) {
-    this.foundation_.open({focusIndex: focusIndex});
+  show() {
+    this.menuSurface_.listen(MDCMenuSurfaceFoundation.strings.OPENED_EVENT, () => {
```

Comment:
This registers a new event listener every time `show()` is called.

It might be better to make it idempotent, so event handlers don't get called more than once for the same event.

And probably add a unit test that calls `show()` several times in a row.